### DISCUSSION TITLE
Zeppelin Masters APE script

### DISCUSSION THREAD

**Discussion by cressidagp** [arcemu/arcemu#438] [Show and tell]
Hi there community! While pandemy at 2020 noticed than Zeppelin Masters wasnt sending the yell message when zeppelin arrived.
Taked a look at plugins and i saw that they arent scripted. So decided to give it a try. 
Did many attemps until finally i get a working script. Its not pro level but will do the job. Retrospectively, i realize now than i should have use AIUpdates intead of RegisterEvent. 
Anyway here is the script and i also started a python version (wich use AIUpdates), but im still figuring out how to proper store and get the variables.

Python: https://pastebin.com/1KAkKq9w

**Comment by dfighter1985** [comment]
Wouldn't it be better if instead of periodically checking for the transport GO there was a script function call when a transport arrives?
Then you could hook into that and make the zeppelin / ship master announce the arrival.

**Comment by cressidagp** [comment]
But i think in that case it will be necesary script the transport gameObject (or i miss something?)... but they cant be scripted. I think they are pushed to world when continent create.

**Reply by dfighter1985** [reply]
Well I could make a server hook for it or some new kind of transport script.

Just checked Arcemu code related to transports and I can see that someone was already thinking about this, just never actually implemented it.

```
		if(mCurrentWaypoint->second.delayed)
		{
			//Transprter Script = sScriptMgr.CreateAIScriptClassForGameObject(GetEntry(), this);
			switch(GetInfo()->DisplayID)
			{
```

**Comment by dfighter1985** [comment]
@cressidagp check the latest commits. I've implemented a new server hook for this.

**Comment by cressidagp** [comment]
Wow! Thanks. That will help a lot! 🤠

**Comment by cressidagp** [comment]
Finally, i get a simple but working version for APE. Updated pastebin.
All Masters are in, but the one from Thunder Bluff may not work since the gameobject its too far away from the creature.
Have fun!
